src/Microsoft.DotNet.XHarness.TestRunners.NUnit/NUnitTestListener.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Xml;
+using Microsoft.DotNet.XHarness.TestRunners.Common;
+using NUnit;
+using NUnit.Engine;
+using NUnit.Framework.Interfaces;
+
+namespace Microsoft.DotNet.XHarness.TestRunners.NUnit
+{
+    internal class NUnitTestListener : ITestEventListener
+    {
+        readonly LogWriter _logger;
+        readonly INUnitTestRunner _runner;
+
+        public NUnitTestListener(INUnitTestRunner runner, LogWriter logger)
+        {
+            _runner = runner ?? throw new ArgumentNullException(nameof (runner));
+            _logger = logger ?? throw new ArgumentNullException(nameof (logger));
+        }
+
+        void TestStarted(XmlNode testEvent)
+        {
+            if (testEvent == null)
+                return;
+
+			if (!string.IsNullOrEmpty (_runner.TestsRootDirectory))
```

Comment:
Do we keep doing this for each test that started? Or this is for the full suit?

Reply:
Full suit.